habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -261,39 +275,91 @@ def reconfigure(self, config: Config):
+            self._targets[target_handle] = mn.Matrix4(
+                [[transform[j][i] for j in range(4)] for i in range(4)]
+            )
+
+    def get_nav_pos(self, pos):
+        pos = mn.Vector3(*pos)
+        height_thresh = 0.15
+        z_min = -0.2
+        use_vs = np.array(self.pathfinder.build_navmesh_vertices())
+
+        if height_thresh is not None:
+            use_vs = use_vs[use_vs[:, 1] < height_thresh]
+        if z_min is not None:
+            use_vs = use_vs[use_vs[:, 2] > z_min]
+        dists = np.linalg.norm(
+            use_vs[:, [0, 2]] - np.array(pos)[[0, 2]], axis=-1
+        )
+
+        closest_idx = np.argmin(dists)
+        return use_vs[closest_idx]
+
     def _recompute_navmesh(self):
-        """Generates the navmesh on the fly. This must be called
+        """Generates the navmesh or loads the saved navmesh if it exists. This must be called
         AFTER adding articulated objects to the scene.
         """
 
-        # cache current motiontype and set to STATIC for inclusion in the NavMesh computation
-        motion_types = []
-        for art_obj in self.art_objs:
-            motion_types.append(art_obj.motion_type)
-            art_obj.motion_type = MotionType.STATIC
-        # compute new NavMesh
-        self.recompute_navmesh(
-            self.pathfinder,
-            self.navmesh_settings,
-            include_static_objects=True,
-        )
-        # optionally save the new NavMesh
-        if self.habitat_config.get("SAVE_NAVMESH", False):
-            scene_name = self.ep_info["scene_id"]
-            inferred_path = scene_name.split(".glb")[0] + ".navmesh"
-            self.pathfinder.save_nav_mesh(inferred_path)
-        # reset cached MotionTypes
-        for art_obj, motion_type in zip(self.art_objs, motion_types):
-            art_obj.motion_type = motion_type
+        scene_name = self.ep_info["scene_id"]
+        navmesh_path = scene_name.split(".glb")[0] + ".navmesh"
+
+        if osp.exists(navmesh_path) and not self.habitat_config.get(
+            "FORCE_RECOMPUTE_NAVMESH", False
+        ):
+            self.pathfinder.load_nav_mesh(navmesh_path)
+        else:
+            # cache current motiontype and set to STATIC for inclusion in the NavMesh computation
+            motion_types = []
+            for art_obj in self.art_objs:
+                motion_types.append(art_obj.motion_type)
+                art_obj.motion_type = MotionType.STATIC
+
+            # compute new NavMesh
+            self.recompute_navmesh(
+                self.pathfinder,
+                self.navmesh_settings,
+                include_static_objects=True,
+            )
+            # optionally save the new NavMesh
+            self.pathfinder.save_nav_mesh(navmesh_path)
+            # reset cached MotionTypes
+            for art_obj, motion_type in zip(self.art_objs, motion_types):
+                art_obj.motion_type = motion_type
+
+    def _get_non_frl_objs(self):
```

Comment:
Probably don't want to pivot on dataset specifics in the long run. If users change the dataset, these checks won't necessarily hold. I assume you are filtering out frl objects (furniture) which were loaded into the scene automatically before dynamic clutter objects.

I'm open to thoughts on how best to address this more generically. Perhaps newly added clutter objects have instance handles cached instead?

Reply:
I actually have no idea what that method was for. It was not used anywhere. I removed it.